lib/entities/quote/DutchQuote.ts

```diff
@@ -277,45 +273,18 @@ export class DutchQuote implements IQuote {
         endAmount: this.amountInEnd,
       });
 
-    if (
-      this.portionRecipient &&
-      this.portionBips &&
-      frontendAndUraEnablePortion(this.request.info.sendPortionEnabled)
-    ) {
-      if (this.request.info.type === TradeType.EXACT_INPUT) {
-        // Amount to swapper
-        builder.output({
-          token: this.tokenOut,
-          startAmount: this.amountOutStart.sub(this.portionAmountOutStart),
-          endAmount: this.amountOutEnd.sub(this.portionAmountOutEnd),
-          recipient: this.request.config.swapper,
-        });
-      } else if (this.request.info.type === TradeType.EXACT_OUTPUT) {
-        // Amount to swapper
-        builder.output({
-          token: this.tokenOut,
-          startAmount: this.amountOutStart,
-          endAmount: this.amountOutEnd,
-          recipient: this.request.config.swapper,
-        });
-      }
-
-      // Amount to portion recipient
-      builder.output({
-        token: this.tokenOut,
-        startAmount: this.portionAmountOutStart,
-        endAmount: this.portionAmountOutEnd,
-        recipient: this.portionRecipient,
-      });
-    } else {
-      // Amount to swapper
-      builder.output({
+    const outputs = getPortionAdjustedOutputs(
+      {
         token: this.tokenOut,
         startAmount: this.amountOutStart,
         endAmount: this.amountOutEnd,
         recipient: this.request.config.swapper,
-      });
-    }
+      },
+      !!this.request.info.sendPortionEnabled,
```

Comment:
Think we need to pass this into `frontendAndUraEnablePortion` ? Since the intended behavior is to only use portion if its passed in the request and enabled in the backedn